### PR TITLE
Robot name

### PR DESCRIPTION
**Pull request by jklemen**

### CODE REVIEW

**Comment by johnmcconnell** [comment]
nice job :+1:

File: robot_name/robot_name.rb [hunk 1]
```diff
@@ -1,32 +1,72 @@
...
+class RobotRegistry
+  attr_reader :registry
+
+  def initialize
+    @registry = Set.new
+  end
+
+  def insert(robot)
+    cautious_insert(robot)
+  end
```
Comment (djkotowski) [inline robot_name/robot_name.rb]: Why have `insert` and `cautious_insert`?

File: robot_name/robot_name.rb [hunk 2]
```diff
@@ -1,32 +1,72 @@
...
+
+  def cautious_insert(robot)
+    unless valid_name?(robot)
+      raise NameCollisionError, 'That robot name is invalid!'
```
Comment (djkotowski) [inline robot_name/robot_name.rb]: Is this really a collision error?

File: robot_name/robot_name.rb [hunk 3]
```diff
@@ -1,32 +1,72 @@
...
+    end
+
+    if registered?(robot)
+      raise NameCollisionError, 'That robot name is already registered!'
+    end
+
```
Comment (djkotowski) [inline robot_name/robot_name.rb]: Don't forget Ruby also lets you use this syntax:
```ruby
raise error if condition
```

File: robot_name/robot_name.rb [hunk 4]
```diff
@@ -1,32 +1,72 @@
...
+    @registry.member?(robot.name)
+  end
+
+  def valid_name?(robot)
+    valid_name = /[[:alpha:]]{2}[[:digit:]]{3}/
+    robot.name =~ valid_name
+  end
+end
+
 class Robot
```
Comment (djkotowski) [inline robot_name/robot_name.rb]: Move this to a separate file.  All files should be named the same as their classes, only using snake case.

**Comment by djkotowski** [comment]
Nice job!  This is the first time I've seen `Set` used instead of an array.

I have a few comments throughout.  No need for a resubmission, though.

File: robot_name/robot_name.rb (hunk 2 above)
Comment (jklemen) [inline robot_name/robot_name.rb]: Nope, that should be a different type of error. Good call.

File: robot_name/robot_name.rb (hunk 1 above)
Comment (jklemen) [inline robot_name/robot_name.rb]: I wan't certain when I started writing it if I wanted a 'blind insert' that simply added to the registry without performing any checks. So I made it a wrapper but in the end forgot to change it.

**Comment by jklemen** [comment]
Thanks!

**Pull request closed without merging**